### PR TITLE
Move getState call to a background thread

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/1202552961248957/1205601953283917/f

### Description
Move `getState` method to a background thread

### Steps to test this PR
Smoke test NetP onboarding flow and feature itself.

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3599** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3599" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3599?utm_source=stack-comment).

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt [hunk 1]
```diff
@@ -58,7 +59,7 @@ class RealNetPWaitlistManager @Inject constructor(
 ) : NetPWaitlistManager {
 
     private val state: MutableStateFlow<NetPWaitlistState> =
-        MutableStateFlow(networkProtectionWaitlist.getState())
+        MutableStateFlow(runBlocking { networkProtectionWaitlist.getState() })
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt:62]: @karlenDimla , I don't really like this `runBlocking`, do you have a better idea?

Comment (CDRussell) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt:62]: I agree that `runBlocking` isn't a good addition here. Could we initialise this quickly with a placeholder value (e.g., `loading`) and launch a coroutine to get the actual value which calls `networkProtectionWaitlist.getState()`?

Comment (aitorvs) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt]: @CDRussell I think I found a good solution now.

**Review by karlenDimla — COMMENTED** [review]
I have been testing this build but I am encountering some flaky behavior with the waitlist state. Will add more details to the task.

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt [hunk 2]
```diff
@@ -55,17 +57,29 @@ class RealNetPWaitlistManager @Inject constructor(
...
     private val dispatcherProvider: DispatcherProvider,
+    @AppCoroutineScope private val coroutineScope: CoroutineScope,
 ) : NetPWaitlistManager {
 
-    private val state: MutableStateFlow<NetPWaitlistState> =
-        MutableStateFlow(networkProtectionWaitlist.getState())
+    // A state flow behaves identically to a shared flow when it is created with the following parameters
+    // See https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-state-flow/
+    // See also https://github.com/Kotlin/kotlinx.coroutines/issues/2515
+    private val state: MutableSharedFlow<NetPWaitlistState> = MutableSharedFlow(
```
Comment (karlenDimla) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt:66]: @aitorvs assuming that it takes awhile before `getState()` emits the value, what would happen to the consumers of this flow?

File: app/src/test/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt [hunk 3]
```diff
@@ -111,9 +110,8 @@ class SettingsViewModelTest {
     fun before() {
         MockitoAnnotations.openMocks(this)
 
-        whenever(networkProtectionWaitlist.getState()).thenReturn(NetPWaitlistState.NotUnlocked)
```
Comment (karlenDimla) [inline app/src/test/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt:114]: @aitorvs also looking into[ SettingsViewmodel](https://github.com/duckduckgo/Android/blob/8c106c192a42d84c8ab7ae67057a5b899afac95a/app/src/main/java/com/duckduckgo/app/settings/SettingsViewModel.kt#L206), `getNetworkProtectionEntryState` which calls `getState()` i s being executed in the main thread. Either we change the flowOn to io or move   networkProtectionWaitlist.getState() somewhere else?

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt (hunk 2 above)
Comment (aitorvs) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPWaitlistManager.kt:66]: Nothing, they'd wait for that value to be emitted.

File: app/src/test/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt (hunk 3 above)
Comment (karlenDimla) [reply app/src/test/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt:114]: Resolved. We `getState` in `io` so the piece of logic in question is no longer relevant.

**Review by karlenDimla — APPROVED** [review]

**Pull request merged**